src/cocol/node/ledger/api.cr

```diff
@@ -63,8 +70,22 @@ post "/blocks/pow" do |env|
 end
 
 post "/blocks/pos" do |env|
-  new_block = Ledger::Block::Pos.from_json(
-    env.request.body.not_nil!)
+  begin
+    new_block = Ledger::Block::Pos.from_json(
+      env.request.body.not_nil!)
```

Comment:
does `not_nil!` convince the compile this will never fail?

Reply:
No it just raises if body is `Nil`
https://crystal-lang.org/api/0.32.1/Object.html#not_nil!-instance-method
https://crystal-lang.org/api/0.32.1/NilAssertionError.html